### DISCUSSION TITLE
Ai-on-the-edge with emoncms

### DISCUSSION THREAD

**Discussion by ote76** [jomjol/AI-on-the-edge-device#2355] [General] [answered]
In the past, I set up a metering system for my gas and electricity meters using the emoncms environment. Unfortunately that  environment is not able to provide a solution for the typical household water meter, so I setup a device based on these excellent resources. The meter just works fine and now I would like to export the water meter values also to the emoncms database, for which I suppose the MQTT interface should be the appropriate link. I have set up mosquitto on a raspberry pi and am also already able to read the various information published by the water meter.
Unfortunately I am a total newbie to mosquitto and am stuck configuring to further publish the water meter data.
Is anyone using a similar setup and has managed to publish AI values to emoncms? Or can guide me to find the right settings?
Thanks so much!

**Comment by caco3** [comment]
Mosquitto is just a MQTT broker. All it does is receiving published topics and forwarding it to any client that subscribed to that topic.
So emoncms would need a built in MQTT client which can be configured to the watermeter topics.
Or you would need to have an extra client which receives the watermeter topics and translates them to whatever emoncms understands.

Overall I can suggest Homeassistant which works great and we provide a direct integration into it.

**Comment by ote76** [comment, marked as answer]
Just for everyone who might experience the same situation, it can be resolved quite easily.
Emoncms comes in two flavours, as a local installation for e.g raspberry pi, or the webversion on emoncms.org.
The first one has two interfaces to upload measurement values to the database, either via a web api or through mqtt communication (which is received by another mechanism to store data).
The emoncms.org server however, can only attach new data through a web api. So in order to store the watermeter values, I set up mosquitto on a raspberry  pi within my home network and also node-red, to capture the values from the mosquitto broker and store them onto emoncms.org through a very simple flow with a http request to make use of the web api.
